<launch>
    <arg name="camera_config" value="MT9V022_MONO_8b_752x480_hdr_64fps.cfg" />
    <!-- Config file in camera_config_files folder of your camera. -->
    <arg name="serial_number" value="" />
    <!-- Serial number of the camera in the form of "XXXX-XXXX-XXXX", leave it blank "" if there is only one camera. -->
    <arg name="horizontal_flip" value="true" />
    <!-- Flip image horizontally. -->
    <arg name="vertical_flip" value="true" />
    <!-- Flip image vertically. -->
    <arg name="frame_id" value="mono_optical_frame" />
    <!--Topic frame_id. -->

    <node name="arducam_camera" pkg="arducam_camera" type="arducam_ros_node" output="screen">
        <!-- type에는 소스파일 확장자 붙이지 않는다 -->
        <param name="config_file" value="$(find arducam_camera)/camera_config_files/$(arg camera_config)" />
        <param name="camera_serial" value="$(arg serial_number)" />
        <param name="horizontal_flip" value="$(arg horizontal_flip)" />
        <param name="vertical_flip" value="$(arg vertical_flip)" />
        <param name="frame_id" value="$(arg frame_id)" />
    </node>
</launch>
